src/core/thread/mle_router.cpp

```diff
@@ -1461,7 +1462,16 @@ void MleRouter::UpdateRoutes(const RouteTlv &aRoute, uint8_t aRouterId)
 
                 if (neighbor->GetLinkQualityOut() != linkQuality)
                 {
+                    uint8_t oldLinkCost = mRouterTable.GetLinkCost(*neighbor);
                     neighbor->SetLinkQualityOut(linkQuality);
+                    nextHop = mRouterTable.GetRouter(neighbor->GetNextHop());
+
+                    // reset MLE advertisement timer if neighbor route cost changed to or from infinite
```

Comment:
update the comment here accordingly as well (for better readability in the future)?

Reply:
The comment already said "route cost" which is accurate. It considers both multihop routing cost and direct link cost.